### PR TITLE
Pe fix fieldfe

### PR DESCRIPTION
**Pull request by Paulie14**
Solves issue #1359.
- fixes `check_compatible_mesh` for discontinuous mesh (resp. simplifies code from `DF_fix_native_io` merge)
- enables reading data from discontinous VTK output (tested in DG)

### CODE REVIEW

File: src/mesh/mesh.h [hunk 1]
```diff
@@ -70,6 +70,23 @@ class BoundarySegment {
     static Input::Type::Record input_type;
 };
 
+class EquivalentMeshMap{
```
Comment (jbrezmorf) [inline src/mesh/mesh.h:73]: Rather change to struct EquivalentMeshMap, in order to:
1. make all public by default
2. emphasize that it is just a pair of maps

File: src/mesh/mesh.h [hunk 2]
```diff
@@ -70,6 +70,23 @@ class BoundarySegment {
...
+public:
```
Comment (jbrezmorf) [inline src/mesh/mesh.h:74]: Add a short comment on the purpose of the structure.

**Review by jbrezmorf — COMMENTED** [review]
Just comment EquivalentMeshMap and create an issue for resolving get_element_data.

File: src/mesh/mesh.h (hunk 1 above)
Comment (Paulie14) [reply src/mesh/mesh.h]: done

File: src/mesh/mesh.h (hunk 2 above)
Comment (Paulie14) [reply src/mesh/mesh.h]: done

**Comment by Paulie14** [comment]
> Just comment EquivalentMeshMap and create an issue for resolving get_element_data.

PR resolved, MSH problems continues in issue #1366 and PR #1369.

**Comment by jbrezmorf** [comment]
check if it pas tests after merging in the new master

**Comment by jbrezmorf** [comment]
Then merge.

**Pull request merged**